### PR TITLE
Remove dialect name from pass option name (part 2)

### PR DESCRIPTION
**Pull request by schweitzpgi**
Remove quake from quake-observe-ansatz. See #771.

### CODE REVIEW

File: include/cudaq/Optimizer/Transforms/Passes.td [hunk 1]
```diff
@@ -327,7 +327,7 @@ def QuakeAddMetadata : Pass<"quake-add-metadata", "mlir::func::FuncOp"> {
   let constructor = "cudaq::opt::createQuakeAddMetadata()";
 }
 
-def QuakeObserveAnsatz : Pass<"quake-observe-ansatz", "mlir::func::FuncOp"> {
+def QuakeObserveAnsatz : Pass<"observe-ansatz", "mlir::func::FuncOp"> {
```
Comment (bmhowe23) [inline include/cudaq/Optimizer/Transforms/Passes.td:330]: If we're changing the pass string name, I'd imagine we'd want to change the pass name (i.e. QuakeObserveAnsatz --> ObserveAnsatz) in order to keep things consistent.

**Review by bmhowe23 — COMMENTED** [review]
If we're going to do this one pass per PR, I'd recommend putting the pass name in the PR so that the commit history will clearly show when various passes were changed. It will also make the auto-generated release notes a bit more descriptive.

File: include/cudaq/Optimizer/Transforms/Passes.td (hunk 1 above)
Comment (schweitzpgi) [reply include/cudaq/Optimizer/Transforms/Passes.td:339]: We can do that later. The name in the code is not a customer facing issue.

**Comment by schweitzpgi** [comment]
> If we're going to do this one pass per PR, I'd recommend putting the pass name in the PR so that the commit history will clearly show when various passes were changed. It will also make the auto-generated release notes a bit more descriptive.

I've updated the commit to reflect the pass or pass collection.

**Review by anthony-santana — APPROVED** [review]

**Pull request merged**